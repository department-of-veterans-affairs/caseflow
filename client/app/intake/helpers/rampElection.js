import { RAMP_INTAKE_STATES } from '../constants';

export const getRampElectionStatus = (rampElection) => {
  if (rampElection.isComplete) {
    return RAMP_INTAKE_STATES.COMPLETED;
  } else if (rampElection.isReviewed) {
    return RAMP_INTAKE_STATES.REVIEWED;
  } else if (rampElection.isStarted) {
    return RAMP_INTAKE_STATES.STARTED;
  }

  return RAMP_INTAKE_STATES.NONE;
};
