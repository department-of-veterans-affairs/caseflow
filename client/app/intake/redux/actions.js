import { ACTIONS } from '../constants';
import ApiUtil from '../../util/ApiUtil';
import { formatDateStringForApi } from '../../util/DateUtil';

export const startNewIntake = () => ({
  type: ACTIONS.START_NEW_INTAKE
});

export const setFileNumberSearch = (fileNumber) => ({
  type: ACTIONS.SET_FILE_NUMBER_SEARCH,
  payload: {
    fileNumber
  }
});

export const doFileNumberSearch = (fileNumberSearch) => (dispatch) => {
  dispatch({
    type: ACTIONS.FILE_NUMBER_SEARCH_START
  });

  return ApiUtil.post('/intake', { data: { file_number: fileNumberSearch } }).
    then(
      (response) => {
        const responseObject = JSON.parse(response.text);

        dispatch({
          type: ACTIONS.FILE_NUMBER_SEARCH_SUCCEED,
          payload: {
            intakeId: responseObject.id,
            name: responseObject.veteran_name,
            formName: responseObject.veteran_form_name,
            fileNumber: responseObject.veteran_file_number
          }
        });
      },
      (error) => {
        const responseObject = JSON.parse(error.response.text);

        dispatch({
          type: ACTIONS.FILE_NUMBER_SEARCH_FAIL,
          payload: {
            errorCode: responseObject.error_code
          }
        });
      }
    );
};

export const setSelectedOption = (optionSelected) => ({
  type: ACTIONS.SET_SELECTED_OPTION,
  payload: {
    optionSelected
  }
});

export const setReceiptDate = (receiptDate) => ({
  type: ACTIONS.SET_RECEIPT_DATE,
  payload: {
    receiptDate
  }
});

export const submitReview = (rampElection) => (dispatch) => {
  dispatch({
    type: ACTIONS.SUBMIT_REVIEW_START
  });

  const data = {
    option_selected: rampElection.optionSelected,
    receipt_date: formatDateStringForApi(rampElection.receiptDate)
  };

  return ApiUtil.patch(`/intake/ramp/${rampElection.intakeId}`, { data }).
    then(
      () => dispatch({ type: ACTIONS.SUBMIT_REVIEW_SUCCEED }),
      () => dispatch({ type: ACTIONS.SUBMIT_REVIEW_FAIL })
    );
};
