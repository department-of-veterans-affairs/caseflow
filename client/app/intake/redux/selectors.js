import { createSelector } from 'reselect';
import { RAMP_INTAKE_STATES } from '../constants';

const getRampElection = (state) => state.rampElection;

export const getRampElectionStatus = createSelector(
  [getRampElection],
  (rampElection) => {
    if (rampElection.isComplete) {
      return RAMP_INTAKE_STATES.COMPLETED;
    } else if (rampElection.isReviewed) {
      return RAMP_INTAKE_STATES.REVIEWED;
    } else if (rampElection.intakeId) {
      return RAMP_INTAKE_STATES.STARTED;
    }

    return RAMP_INTAKE_STATES.NONE;
  }
);
