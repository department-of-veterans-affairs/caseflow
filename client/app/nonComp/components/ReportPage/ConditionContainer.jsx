import React from 'react';
import { useWatch, useFormContext } from 'react-hook-form';
import { ConditionDropdown } from './ConditionDropdown';
import Link from '@department-of-veterans-affairs/caseflow-frontend-toolkit/components/Link';
import { DaysWaiting } from './Conditions/DaysWaiting';
import { DecisionReviewType } from './Conditions/DecisionReviewType';
import { IssueType } from './Conditions/IssueType';
import { IssueDisposition } from './Conditions/IssueDisposition';
import { Facility } from './Conditions/Facility';
import { Personnel } from './Conditions/Personnel';
import PropTypes from 'prop-types';

export const ConditionContainer = ({ control, index, remove }) => {

  const { watch, register } = useFormContext();

  const variableOptions = [
    { label: 'Days Waiting',
      value: 'daysWaiting',
      component: DaysWaiting },
    { label: 'Decision Review Type',
      value: 'decisionReviewType',
      component: DecisionReviewType },
    { label: 'Issue Type',
      value: 'issueType',
      component: IssueType },
    { label: 'Issue Disposition',
      value: 'issueDisposition',
      component: IssueDisposition },
    { label: 'Personnel',
      value: 'personnel',
      component: Personnel },
    { label: 'Facility',
      value: 'facility',
      component: Facility },
  ];

  const determineOptions = () => {
    let conds = watch('conditions');
    let selectedOptions = conds.map((cond) => cond.condition).filter((cond) => cond !== null);

    // personnel and facility are mutually exclusive
    if (selectedOptions.includes('facility')) {
      selectedOptions = selectedOptions.concat('personnel');
    } else if (selectedOptions.includes('personnel')) {
      selectedOptions = selectedOptions.concat('facility');
    }

    return variableOptions.filter((option) =>
      !selectedOptions.some((selectedOption) => option.value === selectedOption));
  };

  const name = `conditions.${index}`;

  const conditionsLength = useWatch({ name: 'conditions' }).length;
  const shouldShowAnd = (conditionsLength > 1) && (index !== (conditionsLength - 1));

  const selectedConditionValue = useWatch({ control, name: `${name}.condition` });

  const getConditionContent = () => {
    const selectedVariableOption = variableOptions.find((opt) => opt.value === selectedConditionValue);

    if (!selectedConditionValue || !selectedVariableOption) {
      return <div></div>;
    }

    if (selectedVariableOption.component) {
      const ConditionContent = selectedVariableOption.component;

      return <ConditionContent {...{ control, register, name }} />;
    }
  };

  return <div className="report-page-segment">
    <div className="cf-app-segment cf-app-segment--alt report-page-variable-condition" >
      <div className="report-page-variable-select">
        <ConditionDropdown {...{ control, determineOptions, name }} />
      </div>
      <div className="report-page-variable-content">{selectedConditionValue ? getConditionContent() : null} </div>
    </div>
    <Link onClick={() => remove(index)}>Remove condition</Link>
    {shouldShowAnd ? <div className="report-page-condition-and">AND</div> : null}
  </div>;
};

ConditionContainer.propTypes = {
  control: PropTypes.object,
  index: PropTypes.number,
  remove: PropTypes.func
};
