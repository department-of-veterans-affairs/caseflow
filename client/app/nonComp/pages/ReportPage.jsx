import React, { useEffect } from 'react';
import { useController, useForm, FormProvider, useFormContext } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { downloadReportCSV } from 'app/nonComp/actions/changeHistorySlice';
import { css } from 'glamor';
import PropTypes from 'prop-types';

import Button from 'app/components/Button';
import NonCompLayout from '../components/NonCompLayout';
import { conditionsSchema, ReportPageConditions } from '../components/ReportPage/ReportPageConditions';

import { yupResolver } from '@hookform/resolvers/yup';
import * as yup from 'yup';
import { fetchUsers } from 'app/nonComp/actions/usersSlice';

import RHFControlledDropdownContainer from 'app/nonComp/components/ReportPage/RHFControlledDropdown';
import { timingSchema, TimingSpecification } from 'app/nonComp/components/ReportPage/TimingSpecification';

import Checkbox from 'app/components/Checkbox';
import RadioField from 'app/components/RadioField';

import { get } from 'lodash';

import {
  REPORT_TYPE_OPTIONS,
  RADIO_EVENT_TYPE_OPTIONS,
  RADIO_STATUS_OPTIONS,
  RADIO_STATUS_REPORT_TYPE_OPTIONS,
  SPECIFIC_STATUS_OPTIONS,
  SPECTIFIC_EVENT_OPTIONS
} from 'constants/REPORT_TYPE_CONSTANTS';
import * as ERRORS from 'constants/REPORT_PAGE_VALIDATION_ERRORS';

const buttonInnerContainerStyle = css({
  display: 'flex',
  gap: '32px',
});

const buttonOuterContainerStyling = css({
  display: 'flex',
  justifyContent: 'space-between',
  marginTop: '4rem',
});

const specificEventTypeSchema = yup.lazy((value) => {
  // eslint-disable-next-line no-undefined
  if (value === undefined) {
    return yup.mixed().notRequired();
  }

  return yup.object({
    added_decision_date: yup.boolean(),
    added_issue: yup.boolean(),
    added_issue_no_decision_date: yup.boolean(),
    claim_created: yup.boolean(),
    claim_closed: yup.boolean(),
    claim_status_incomplete: yup.boolean(),
    claim_status_inprogress: yup.boolean(),
    completed_disposition: yup.boolean(),
    removed_issue: yup.boolean(),
    withdrew_issue: yup.boolean(),
  }).test('at-least-one-true', ERRORS.AT_LEAST_ONE_OPTION, (obj) => {
    return Object.values(obj).some((val) => val === true);
  });
});

const specificStatusSchema = yup.lazy((value) => {
  // eslint-disable-next-line no-undefined
  if (value === undefined) {
    return yup.mixed().notRequired();
  }

  return yup.object({
    incomplete: yup.boolean(),
    in_progress: yup.boolean(),
    completed: yup.boolean(),
  }).test('at-least-one-true', ERRORS.AT_LEAST_ONE_OPTION, (obj) => {
    return Object.values(obj).some((val) => val === true);
  });
});

const schema = yup.object().shape({
  conditions: conditionsSchema,
  timing: timingSchema,
  specificEventType: specificEventTypeSchema,
  specificStatus: specificStatusSchema
});

const ReportPageButtons = ({
  history,
  isGenerateButtonDisabled,
  handleClearFilters,
  handleSubmit,
  loading,
  loadingText }) => {
  return (

    <div {...buttonOuterContainerStyling}>
      <Button
        classNames={['cf-modal-link', 'cf-btn-link']}
        label="cancel-report"
        name="cancel-report"
        onClick={() => history.push('/vha')}
      >
        Cancel
      </Button>
      <div {...buttonInnerContainerStyle}>
        <Button
          classNames={['usa-button-secondary']}
          label="clear-filters"
          name="clear-filters"
          onClick={handleClearFilters}
          disabled={isGenerateButtonDisabled}
        >
          Clear filters
        </Button>
        <Button
          classNames={['usa-button']}
          label="generate-report"
          name="generate-report"
          onClick={handleSubmit}
          disabled={isGenerateButtonDisabled}
          loading={loading}
          loadingText={loadingText}
        >
          Generate task report
        </Button>
      </div>
    </div>
  );
};

const RHFCheckboxGroup = ({ options, name, control }) => {
  const { field } = useController({
    control,
    name
  });

  const { errors } = useFormContext();

  const [value, setValue] = React.useState({});

  let fieldClasses = 'checkbox';

  const errorMessage = get(errors, name)?.message;

  if (errorMessage) {
    fieldClasses += ' usa-input-error';
    fieldClasses += ' less-error-padding';
  }

  return (
    <fieldset className={fieldClasses} style={{ paddingLeft: '30px' }}>
      {errorMessage ? <div className="usa-input-error-message">{ errorMessage }</div> : null}
      {options.map((option) => (
        <div key={option.id}>
          <Checkbox
            name={`${name}.${option.id}`}
            key={`${name}.${option.id}`}
            label={option.label}
            stronglabel
            onChange={(val) => {
              value[option.id] = val;
              field.onChange(value);
              setValue(value);
            }}
            unpadded
          />
        </div>
      ))}
    </fieldset>
  );
};

const RHFRadioButton = ({ options, name, control, label }) => {
  const { field } = useController({
    control,
    name,
  });

  return (
    <div style={{ marginTop: '20px' }}>
      <RadioField
        name=""
        label={label}
        vertical
        options={options}
        stronglabel
        value={field.value}
        onChange={(val) => {
          field.onChange(val);
        }}
      />
    </div>
  );
};

const ReportPage = ({ history }) => {
  const defaultFormValues = {
    reportType: '',
    conditions: [],
    timing: {
      range: null,
      startDate: '',
      endDate: '',
    },
    radioEventAction: 'all_events_action',
    radioStatus: 'all_statuses',
    radioStatusReportType: 'last_action_taken',
    specificStatus: {
      incomplete: '',
      in_progress: '',
      completed: '',
      cancelled: ''
    },
    specificEventType: {
      added_decision_date: '',
      added_issue: '',
      added_issue_no_decision_date: '',
      claim_created: '',
      claim_closed: '',
      claim_status_incomplete: '',
      claim_status_inprogress: '',
      completed_disposition: '',
      removed_issue: '',
      withdrew_issue: '',
    }
  };

  const methods = useForm({
    defaultValues: { ...defaultFormValues },
    resolver: yupResolver(schema),
    mode: 'onSubmit',
    reValidateMode: 'onSubmit'
  });

  const { reset, watch, formState, control, handleSubmit } = methods;
  const dispatch = useDispatch();
  const businessLineUrl = useSelector((state) => state.nonComp.businessLineUrl);
  const csvGeneration = useSelector((state) => state.changeHistory.status);
  const isCSVGenerating = csvGeneration === 'loading';
  const watchReportType = watch('reportType');
  const watchRadioEventAction = watch('radioEventAction');
  const watchRadioStatus = watch('radioStatus');

  const processConditionOptions = (condition, options) => {
    let formattedOptions;

    switch (condition) {
    case 'decisionReviewType':
      formattedOptions = Object.keys(options).filter((key) => options[key]);
      break;
    // Multi select conditions
    case 'personnel':
    case 'facility':
    case 'issueDisposition':
    case 'issueType':
      formattedOptions = Object.values(options)[0].map((item) => item.value);
      break;
    // Else it is probably already an object, so it just pass the existing options
    default:
      formattedOptions = options;
    }

    return formattedOptions;
  };

  const parseFilters = (data) => {
    const filters = {};

    // Add report type to the filter
    filters.reportType = data.reportType;

    // Event specific event types to the filter
    if (data.radioEventAction === 'specific_events_action') {
      filters.events = Object.keys(data.specificEventType).filter((key) => data.specificEventType[key] === true);
    }

    // Add specific status types to the filter
    if (data.radioStatus === 'specific_status') {
      filters.statuses = Object.keys(data.specificStatus).filter((key) => data.specificStatus[key] === true);
    }

    // Add timing to the filter
    filters.timing = data.timing;

    // Add Status report type to the filter
    filters.statusReportType = data.radioStatusReportType;

    // Parse all the Conditions and add them to the filter
    const transformedConditions = data?.conditions?.reduce((result, item) => {
      const { condition, options } = item;

      if (condition && options) {
        // Parse the individual conditions
        const newOptions = processConditionOptions(condition, options);

        result[condition] = newOptions;
      }

      return result;
    }, {});

    // Add the all the parsed conditions into the filters
    Object.assign(filters, transformedConditions);

    return filters;
  };

  const submitForm = (data) => {
    const filterData = parseFilters(data);

    // Generate and trigger the download of the CSV
    dispatch(downloadReportCSV({ organizationUrl: businessLineUrl, filterData: { filters: filterData } }));
  };

  useEffect(() => {
    dispatch(fetchUsers({ queryType: 'organization', queryParams: { query: 'vha' } }));
  }, []);

  return (
    <NonCompLayout
      buttons={
        <ReportPageButtons
          history={history}
          isGenerateButtonDisabled={!formState.isDirty}
          handleClearFilters={() => reset()}
          handleSubmit={handleSubmit(submitForm)}
          loading={isCSVGenerating}
          loadingText="Generating CSV"
        />
      }
    >
      <h1>Generate task report</h1>
      <FormProvider {...methods}>
        <form>
          <RHFControlledDropdownContainer
            header="Type of report"
            name="reportType"
            label="Report Type"
            options={REPORT_TYPE_OPTIONS}
          />
          {watchReportType === 'status' ? (<>
            <RHFRadioButton
              options={RADIO_STATUS_OPTIONS}
              methods={methods}
              name="radioStatus"
            />
            {watchRadioStatus === 'specific_status' ? (
              <RHFCheckboxGroup
                options={SPECIFIC_STATUS_OPTIONS}
                control={control}
                name="specificStatus"
              />) :
              null
            }
            <RHFRadioButton
              options={RADIO_STATUS_REPORT_TYPE_OPTIONS}
              methods={methods}
              label="Select type of status report"
              name="radioStatusReportType"
            />
          </>) :
            null
          }
          {watchReportType === 'event_type_action' ? (
            <RHFRadioButton
              options={RADIO_EVENT_TYPE_OPTIONS}
              methods={methods}
              name="radioEventAction"
            />
          ) : null
          }
          {watchReportType === 'event_type_action' &&
          watchRadioEventAction === 'specific_events_action' ? (
              <RHFCheckboxGroup
                options={SPECTIFIC_EVENT_OPTIONS}
                control={control}
                name="specificEventType"
              />
            ) : null
          }
          {watchReportType === 'event_type_action' ?
            <TimingSpecification /> :
            null
          }
          {formState.isDirty ? <ReportPageConditions /> : null}
        </form>
      </FormProvider>
    </NonCompLayout>
  );
};

ReportPageButtons.propTypes = {
  history: PropTypes.object,
  isGenerateButtonDisabled: PropTypes.bool,
  handleClearFilters: PropTypes.func,
  handleSubmit: PropTypes.func,
  loading: PropTypes.bool,
  loadingText: PropTypes.string,
};

ReportPage.propTypes = {
  history: PropTypes.object,
};

RHFCheckboxGroup.propTypes = {
  options: PropTypes.array,
  control: PropTypes.object,
  name: PropTypes.string,
  errorMessage: PropTypes.string
};

RHFRadioButton.propTypes = {
  options: PropTypes.array,
  control: PropTypes.object,
  label: PropTypes.string,
  name: PropTypes.string
};

export default ReportPage;
