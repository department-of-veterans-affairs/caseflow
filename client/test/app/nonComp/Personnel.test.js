import React from 'react';
import { axe } from 'jest-axe';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore, compose } from 'redux';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import selectEvent from 'react-select-event';

import ReportPage from 'app/nonComp/pages/ReportPage';
import { getVhaUsers } from 'test/helpers/reportPageHelper';
import CombinedNonCompReducer from 'app/nonComp/reducers';

const setup = (storeValues = {}) => {
  const store = createStore(
    CombinedNonCompReducer,
    storeValues,
    compose(applyMiddleware(thunk))
  );

  return render(
    <Provider store={store}>
      <ReportPage />
    </Provider>
  );
};

const navigateToPersonnel = async () => {
  await selectEvent.select(screen.getByLabelText('Report Type'), ['Status', 'Event / Action']);

  const addConditionBtn = screen.getByText('Add Condition');

  userEvent.click(addConditionBtn);

  const selectText = screen.getByText('Select a variable');

  await selectEvent.select(selectText, ['Personnel']);
};

beforeEach(() => {
  getVhaUsers();
});

describe('Personnel', () => {
  describe('component renders correctly', () => {
    it('passes a11y testing', async () => {
      const { container } = setup();

      await navigateToPersonnel();

      const results = await axe(container);

      expect(results).toHaveNoViolations();
    });

    it('renders correctly', async () => {
      const { container } = setup();

      await navigateToPersonnel();

      expect(container).toMatchSnapshot();
    });
  });

  describe('component functions correctly', () => {
    beforeEach(async () => {
      setup();
      await navigateToPersonnel();
    });

    const selectPlaceholder = 'Select...';
    const teamMember1 = 'VHAUSER01';

    it('renders a dropdown with the correct label', async () => {
      expect(screen.getByText('VHA team members')).toBeInTheDocument();
      expect(screen.getAllByText(selectPlaceholder).length).toBe(2);
    });

    it('allows to select multiple options from dropdown', async () => {
      let selectText = screen.getAllByText(selectPlaceholder);

      await selectEvent.select(selectText[1], [teamMember1]);

      selectText = screen.getByText(teamMember1);
      const teamMember2 = 'VHAUSER02';

      await selectEvent.select(selectText, [teamMember2]);

      expect(screen.getByText(teamMember1)).toBeInTheDocument();
      expect(screen.getByText(teamMember2)).toBeInTheDocument();
    });

    it('selects an option from dropdown, then removes it and renders an error', async () => {
      const selectText = screen.getAllByText(selectPlaceholder);

      await selectEvent.select(selectText[1], [teamMember1]);
      expect(screen.getByText(teamMember1)).toBeInTheDocument();

      const clearBtn = document.querySelector('.cf-select__indicator.cf-select__clear-indicator');

      userEvent.click(clearBtn);

      expect(screen.queryByText(teamMember1)).not.toBeInTheDocument();
    });
  });
});
