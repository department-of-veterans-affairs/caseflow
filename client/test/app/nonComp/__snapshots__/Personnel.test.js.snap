// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personnel component renders correctly renders correctly 1`] = `
<div>
  <div>
    <section
      class="cf-app-segment cf-app-segment--alt"
    >
      <div>
        <h1>
          Generate task report
        </h1>
        <form>
          <h2>
            Type of report
          </h2>
          <div
            class=""
          >
            <div
              class="cf-form-dropdown dropdown-reportType"
            >
              <label
                class="question-label"
                for="report-type"
                id="report-type-label"
              >
                <span>
                  Report Type
                </span>
              </label>
              <div
                class="cf-select"
              >
                <div
                  class=" css-2b097c-container"
                >
                  <div
                    class="cf-select__control css-yk16xz-control"
                  >
                    <div
                      class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                    >
                      <div
                        class="cf-select__single-value css-1uccc91-singleValue"
                      >
                        Event / Action
                      </div>
                      <div
                        class="css-81nyic-Input"
                      >
                        <div
                          class="cf-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-labelledby="report-type-label"
                            aria-owns="report-type-listbox"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="report-type"
                            role="combobox"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 0; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-6q0nyr-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="reportType"
                    type="hidden"
                    value="event_type_action"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            style="margin-top: 20px;"
          >
            <fieldset
              class="usa-fieldset-inputs cf-form-radio "
            >
              <legend
                class=""
              >
                <span>
                  
                   
                </span>
              </legend>
              <div
                class="cf-form-radio-options"
              >
                <div
                  class="cf-form-radio-option"
                >
                  <input
                    checked=""
                    id="_all_events_action"
                    name=""
                    type="radio"
                    value="all_events_action"
                  />
                  <label
                    class=""
                    for="_all_events_action"
                  >
                    All Events / Actions
                  </label>
                </div>
                <div
                  class="cf-form-radio-option"
                >
                  <input
                    id="_specific_events_action"
                    name=""
                    type="radio"
                    value="specific_events_action"
                  />
                  <label
                    class=""
                    for="_specific_events_action"
                  >
                    Specific Events / Actions
                  </label>
                </div>
              </div>
            </fieldset>
          </div>
          <div>
            <hr
              style="margin-top: 50px; margin-bottom: 50px;"
            />
            <h2>
              Timing specifications
            </h2>
            <div
              class=""
            >
              <div
                class="cf-form-dropdown dropdown-timing.range"
              >
                <label
                  class="question-label"
                  for="timing-range"
                  id="timing-range-label"
                >
                  <span>
                    Range
                    <span
                      class="cf-optional"
                    >
                      Optional
                    </span>
                  </span>
                </label>
                <div
                  class="cf-select"
                >
                  <div
                    class=" css-2b097c-container"
                  >
                    <div
                      class="cf-select__control css-yk16xz-control"
                    >
                      <div
                        class="cf-select__value-container css-g1d714-ValueContainer"
                      >
                        <div
                          class="cf-select__placeholder css-1wa3eu0-placeholder"
                        >
                          Select...
                        </div>
                        <div
                          class="css-81nyic-Input"
                        >
                          <div
                            class="cf-select__input"
                            style="display: inline-block;"
                          >
                            <input
                              aria-autocomplete="list"
                              aria-expanded="false"
                              aria-haspopup="true"
                              aria-labelledby="timing-range-label"
                              aria-owns="timing-range-listbox"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              id="timing-range"
                              role="combobox"
                              spellcheck="false"
                              style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                        />
                        <div
                          aria-hidden="true"
                          class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="css-6q0nyr-Svg"
                            focusable="false"
                            height="20"
                            viewBox="0 0 20 20"
                            width="20"
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <input
                      name="timing.range"
                      type="hidden"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div>
            <hr
              style="margin: 50px 0px;"
            />
            <h2
              style="margin: 0px;"
            >
              Conditions
            </h2>
            <div
              class="report-page-segment"
            >
              <div
                class="cf-app-segment cf-app-segment--alt report-page-variable-condition"
              >
                <div
                  class="report-page-variable-select"
                >
                  <div
                    class=""
                  >
                    <div
                      class="cf-form-dropdown dropdown-conditions.0.condition"
                    >
                      <label
                        class="question-label"
                        for="conditions-0-condition"
                        id="conditions-0-condition-label"
                      >
                        <span>
                          Variable
                        </span>
                      </label>
                      <div
                        class="cf-select"
                      >
                        <div
                          class="cf-select--is-disabled css-14jk2my-container"
                        >
                          <div
                            class="cf-select__control cf-select__control--is-disabled css-1fhf3k1-control"
                          >
                            <div
                              class="cf-select__value-container cf-select__value-container--has-value css-g1d714-ValueContainer"
                            >
                              <div
                                class="cf-select__single-value cf-select__single-value--is-disabled css-107lb6w-singleValue"
                              >
                                Personnel
                              </div>
                              <input
                                aria-autocomplete="list"
                                class="css-62g3xt-dummyInput"
                                disabled=""
                                id="conditions-0-condition"
                                readonly=""
                                tabindex="0"
                                value=""
                              />
                            </div>
                            <div
                              class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                            >
                              <span
                                class="cf-select__indicator-separator css-109onse-indicatorSeparator"
                              />
                              <div
                                aria-hidden="true"
                                class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="css-6q0nyr-Svg"
                                  focusable="false"
                                  height="20"
                                  viewBox="0 0 20 20"
                                  width="20"
                                >
                                  <path
                                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="report-page-middle-content"
                >
                  including
                </div>
                <div
                  class="report-page-variable-content"
                >
                  <div
                    class="report-page-multi-select-dropdown"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="cf-form-dropdown dropdown-conditions.0.options.personnel"
                      >
                        <label
                          class="question-label"
                          for="conditions-0-options-personnel"
                          id="conditions-0-options-personnel-label"
                        >
                          <span>
                            VHA team members
                          </span>
                        </label>
                        <div
                          class="cf-select"
                        >
                          <div
                            class=" css-2b097c-container"
                          >
                            <div
                              class="cf-select__control css-yk16xz-control"
                            >
                              <div
                                class="cf-select__value-container cf-select__value-container--is-multi css-g1d714-ValueContainer"
                              >
                                <div
                                  class="cf-select__placeholder css-1wa3eu0-placeholder"
                                >
                                  Select...
                                </div>
                                <div
                                  class="css-81nyic-Input"
                                >
                                  <div
                                    class="cf-select__input"
                                    style="display: inline-block;"
                                  >
                                    <input
                                      aria-autocomplete="list"
                                      aria-expanded="false"
                                      aria-haspopup="true"
                                      aria-labelledby="conditions-0-options-personnel-label"
                                      aria-owns="conditions-0-options-personnel-listbox"
                                      autocapitalize="none"
                                      autocomplete="off"
                                      autocorrect="off"
                                      id="conditions-0-options-personnel"
                                      role="combobox"
                                      spellcheck="false"
                                      style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                      tabindex="0"
                                      type="text"
                                      value=""
                                    />
                                    <div
                                      style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                                    />
                                  </div>
                                </div>
                              </div>
                              <div
                                class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                              >
                                <span
                                  class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                                />
                                <div
                                  aria-hidden="true"
                                  class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class="css-6q0nyr-Svg"
                                    focusable="false"
                                    height="20"
                                    viewBox="0 0 20 20"
                                    width="20"
                                  >
                                    <path
                                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                    />
                                  </svg>
                                </div>
                              </div>
                            </div>
                            <div>
                              <input
                                name="conditions.0.options.personnel"
                                type="hidden"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <a>
                Remove condition
              </a>
            </div>
            <span>
              <button
                class="cf-submit usa-button"
                type="button"
              >
                Add Condition
              </button>
            </span>
          </div>
        </form>
      </div>
    </section>
    <div
      data-css-1rfrm1q=""
    >
      <span>
        <button
          class="cf-modal-link cf-btn-link usa-button"
          id="button-cancel-report"
          type="button"
        >
          Cancel
        </button>
      </span>
      <div
        data-css-cfdlna=""
      >
        <span>
          <button
            class="usa-button-secondary usa-button"
            id="button-clear-filters"
            type="button"
          >
            Clear filters
          </button>
        </span>
        <span>
          <button
            class="usa-button usa-button"
            id="button-generate-report"
            type="button"
          >
            Generate task report
          </button>
        </span>
      </div>
    </div>
  </div>
</div>
`;
